transit_odp/publish/views/timetable/seasonal_services.py

```diff
@@ -4,22 +4,53 @@
 from django.forms import Form
 from django.http import HttpResponseRedirect
 from django.utils.translation import gettext as _
+from django.views.generic.edit import BaseDeleteView
 from django_hosts import reverse
 from django_tables2 import SingleTableView
+from django_tables2.paginators import Paginator
 from formtools.wizard.views import SessionWizardView
 
 import config.hosts
 from transit_odp.common.view_mixins import BODSBaseView
+from transit_odp.common.views import BaseUpdateView
 from transit_odp.organisation.models.data import SeasonalService
 from transit_odp.organisation.models.organisations import Organisation
-from transit_odp.publish.forms import (
+from transit_odp.publish.forms.constants import get_table_page
+from transit_odp.publish.forms.seasonal_services import (
     SeasonalServiceEditDateForm,
+    SeasonalServiceEditRegistrationCodeDateForm,
     SeasonalServiceLicenceNumberForm,
 )
 from transit_odp.timetables.tables import SeasonalServiceTable
+from transit_odp.users.constants import DATASET_MANAGE_TABLE_PAGINATE_BY
 from transit_odp.users.views.mixins import OrgUserViewMixin
 
 
+class SeasonalServiceView(OrgUserViewMixin, SingleTableView):
+    template_name = "publish/seasonal_services/index.html"
+    model = SeasonalService
+    table_class = SeasonalServiceTable
+    paginate_by = DATASET_MANAGE_TABLE_PAGINATE_BY
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                "pk1": self.org_id,
+                "organisation": Organisation.objects.get(id=self.org_id),
+                "seasonal_services_counter": SeasonalService.objects.get_count_in_organisation(
+                    self.org_id
+                ),
+            }
+        )
+        return context
+
+    def get_queryset(self):
+        self.org_id = self.kwargs["pk1"]
+        qs = super().get_queryset()
+        return qs.filter(licence__organisation_id=self.org_id).add_registration_number()
```

Comment:
Dont forget to order_by end data for two reasons:
  1) Its in the requirements
  2) Without an order by clause this queryset will return unordered results, there is actually a warning about this in the logs when the endpoint is hit. Unordered means your page number work is useless because there is no guarantee which page a seasonal service will be on.

Reply:
I though was unordered AND the requirements was a typo because in the invision app the multiple page is unordered